sql/sql_connect.cc

```diff
@@ -1000,6 +1000,28 @@ static int check_connection(THD *thd)
     bzero((char*) &net->vio->remote, sizeof(net->vio->remote));
   }
   vio_keepalive(net->vio, TRUE);
+
+  if (global_system_variables.keepalive_probes != 9)
```

Comment:
a way to set keepalive interval and timeout on Windows, per socket , is here https://msdn.microsoft.com/en-us/library/windows/desktop/dd877220(v=vs.85).aspx . 
On OSX, TCP_KEEPALIVE sets the interval https://stackoverflow.com/questions/15860127/how-to-configure-tcp-keepalive-under-mac-os-x . The probes are nowhere but on Linux, as I said